### ISSUE TITLE
http:Request strips newline from string payload when using .setTextPayload(string)

### ISSUE THREAD

**Issue by wyldbill** [ballerina-platform/ballerina-lang#15514]
**Description:**
When using http:Request.setTextPayload(), at least one newline (\n) is stripped from the supplied payload string. I don't know if other whitespace is affected.

This was discovered by using the http:Client to insert _bulk records into ElasticSearch. The _bulk call requires a payload of json objects delimited by newlines. The actual body is not valid json, so setJsonPayload() cannot be used, and the Content-Type must be set manually to application-json.

The workaround is to convert the string to a byte[] and use setBinaryPayload(), explicitly setting the Content-type.

**Steps to reproduce:**
Create or obtain an ElasticSearch server.
create the payload:
```
json header = {"index":{"_index":"foo", "_type":"foo", "_id":1234}}
json data = {"foo":"bar", "baz":"qux"}
string payload = header.toString() + "\n" + data.toString() + "\n"; //note trailing new line
```
send the request
```
http:Client clientEndpoint = new(<your es endpoint>)
http:Request req = new;
req.setTextPayload(payload,contentType = "application/json");
ClientEndpoint->post("/_bulk", req);
```
This fails with an error similar to:
```
{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"The bulk request must be terminated by a newline [\n]"}],"type
```
using the binary payload
```    
byte[] bytes = payload.toByteArray("UTF-8");
req.setBinaryPayload(bytes,contentType = "application/json");
ClientEndpoint->post("/_bulk", req);
```
works as expected.

**Affected Versions:**
0.991.0

**OS, DB, other environment details and versions:**
Windows 10
ElasticSearch 6.7 as AWS Service

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

**Comment by wggihan** [comment]
@wyldbill thanks for reporting this issue. We are planning to do a next release on end of July and we will try to address this.

**Comment by riyafa** [comment]
Fixed in #18082
